Source/CombatExtended/Compatibility/Race AutoPatchers/RaceAutoPatcher.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+using RimWorld;
+
+namespace CombatExtended.Compatibility
+{
+    [StaticConstructorOnStartup]
+    public class RaceAutoPatcher
+    {
+        static RaceAutoPatcher()
+        {
+
+            #region Animal patching
+            var animalsUnpatched = DefDatabase<ThingDef>.AllDefs.Where(x =>
+                   x.race != null &&
+                   x.race.Animal &&
+                   x.tools.Any(y => !(y is ToolCE))
+                );
+
+
+            foreach (ThingDef animal in animalsUnpatched)
+            {
+                Log.Message("Patching " +  animal.defName.Colorize(UnityEngine.Color.green));
+
+
+                var newTools = new List<Tool>();
+
+
+                #region Tool patching
+                foreach (Tool tool in animal.tools)
+                {
+                    if ( !(tool is ToolCE) )

```

Comment:
because animal.tools is assigned to `newTools`, you should add existing tools which are already `ToolCe`'s, otherwise they just get removed